src/lib/mongoBackend/mongoGetSubscription.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+
+#include "common/sem.h"
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+#include "mongoBackend/mongoGetSubscription.h"
+#include "mongoBackend/MongoGlobal.h"
+#include "mongoBackend/collectionOperations.h"
+
+#include "mongo/client/dbclient.h"
+
+using namespace ngsiv2;
+
+
+/* ****************************************************************************
+*
+* mongoGetSubscription -
+*/
+OrionError mongoGetSubscription
```

Comment:
Interesting touch to make mongoBackend return the entire OrionError struct.
Better option than the SccOk that is always returned right now ...

However, this is a entire object ...

Would this work:

    const OrionError& mongoGetSubscription

?
I mean, will the assignment work for the caller of this function?

An alternative would to add a pointer to the OrionError structure as a parameter